purchase_order_line_deep_sort/models/purchase_order.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2018 Tecnativa - Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from odoo import api, fields, models
+from .res_company import SORTING_CRITERIA, SORTING_DIRECTION
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    line_order = fields.Selection(
+        selection=SORTING_CRITERIA,
+        string='Sort Lines By',
+        default=lambda self: self.env.user.company_id.default_po_line_order,
+    )
+    line_direction = fields.Selection(
+        selection=SORTING_DIRECTION,
+        string='Sort Direction',
+        default=lambda self:
+            self.env.user.company_id.default_po_line_direction,
+    )
+
+    @api.onchange('line_order')
+    def onchange_line_order(self):
+        if not self.line_order:
+            self.line_direction = False
+
+    @api.multi
+    def _sort_purchase_line(self):
+        if not self.line_order and not self.line_direction:
+            return
+        order = self.line_order
+        reverse = self.line_direction == 'desc'
+        sequence = 0
+        key = eval("lambda p: p.%s" % order)
```

Comment:
Wouldn't it be better to use Odoo's `safe_eval`?

Reply:
This is not Python code, and I prefer to not have an opened door for any kind of expression.

Reply:
I mean: this is coming from some pre-defined values in selection field, and doing `safe_eval` is not going to work because lambda expressions are forbidden there, but the scope of this is totally limited.

Reply:
I'm seeing any way that we should silence the linter. Let me check alternatives.

Reply:
I have used another safer technique for this.